### ISSUE TITLE
Spell check of meaningful text inside commands (e.g. moderncv cventry{TEXT}) 

### ISSUE THREAD

**Issue by FranzFelberer** [texstudio-org/texstudio#260]
### Environment
- TeXstudio: TeXstudio 2.12.10 (git 2.12.10)
- Qt: Qt-Version 5.11.1, kompiliert mit Qt 5.11.1 R
- OS: Mac 10.13.6 (17G65)
- TeX distribution: texlive

### Expected behavior
Spell check every meaningful text, that is shown in the output file

### Actual behavior
ignoring text entries for \cventry arguments that are displayed

### How to reproduce
open moderncv example from github
https://github.com/xdanaux/moderncv

**Comment by sunderme** [comment]
txs uses the cwl files to determine the meaning of the arguments. Automatically scanned packages don‘t have that information and hence all arguments are considered to be non-text arguments. To change that you need to change the cwl file, see wiki and manual 
http://texstudio.sourceforge.net/manual/current/usermanual_en.html#CWLDESCRIPTION

**Comment by sunderme** [comment]
I don‘t understand your first issue. It completes with the command which you want ?

2) the argument needs to be „text“ or „title“

**Comment by FranzFelberer** [comment]
sorry for the confusion. I didn't make myself very clear:
1. my moderncv.cwl file is ignored by txs. Any idea why?

2. so the arguments should like below in order to spellcheck the second argument and use the text between % % for auto-completion?

\cventry{%<year%>--%<year%>}{text%<degree or job title%>}

**Comment by FranzFelberer** [comment]
Doing what you mentioned solves the problem that the "manuell" spell checker checks this text, I was referring to the on-type spell checker that indicates mistakes while writing by a red unline-ripple. It shows this behavior if in the same argument the text has a property like \testbf.
See image: It indicates no error for the first "wrongwriten" but for the second which is in a \textbf environment.

![image](https://user-images.githubusercontent.com/20334762/43365865-af92acba-9333-11e8-9236-1113371b7973.png)

**Comment by sunderme** [comment]
`\cventry{year-year}{degree or job title%text}{institution or employer%text}{city}{grade}{description%text}#n` works fine here

**Comment by FranzFelberer** [comment]
Thank you, I got it working, but I experience some other strange behavior.

I defined a command in my tex file:
```
% switch between languages
\newcommand{\multilang}[3]{#1}	% #1 en, #2 fr, #3 de
```

In ~/.config/texstudio/completion/usercommands.cwl I add the line:
`\multilang{en%text}{fr%text}{de%text}#n`

But completion results in:
`\multilang{arg1}{arg2}{arg3}`

Strange is, that it was working and now it doesn't anymore.

**Comment by sunderme** [comment]
define the command in a sty file, and it should work

**Comment by stefanbschneider** [comment]
The [`class-moderncv.cwl`](https://github.com/texstudio-org/texstudio/blob/master/completion/class-moderncv.cwl) looks like this issue should be fixed.

But I still have the problem that TeXStudio doesn't do spellchecking in my `\cventry` commands. What's the problem?

**Comment by sunderme** [comment]
do you have the latest version of txs ?

**Comment by stefanbschneider** [comment]
Yes, version 2.12.14

**Comment by stefanbschneider** [comment]
Here's a screenshot:

![image](https://user-images.githubusercontent.com/28340802/55474305-1055c500-5611-11e9-804b-1195cf3a5107.png)

**Comment by sunderme** [comment]
Here is mine:
![test](https://user-images.githubusercontent.com/14033169/55492343-5cb3fb80-5637-11e9-9acb-563371de25b8.png)

**Comment by sunderme** [comment]
you need to use a development build to make use of the fix

**Comment by stefanbschneider** [comment]
So this is a bug, which is not yet fixed in the current version 2.12.14?

**Comment by sunderme** [comment]
looks like it
